src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -39,9 +40,23 @@ public static function register($mode = 0)
         if (self::$isRegistered) {
             return;
         }
-        if (self::MODE_WEAK !== $mode && (!isset($mode[0]) || '/' !== $mode[0])) {
-            $mode = preg_match('/^[1-9][0-9]*$/', $mode) ? (int) $mode : 0;
-        }
+
+        $mode = function () use ($mode) {
```

Comment:
reusing the same variable name to store the function makes the code much harder to read